File: voldemort/scala/create/voldemort_create.scala
// ---------------------------------------------------------------------
//
//	voldemort_create.scala
//
//						Sep/09/2013
//
// ---------------------------------------------------------------------
import voldemort.client.ClientConfig
import voldemort.client.SocketStoreClientFactory
import voldemort.client.StoreClient
import voldemort.versioning.Versioned

import scala.collection.mutable
// ---------------------------------------------------------------------
object voldemort_create
{
// -------------------------------------------------------------------
def data_prepare_proc ():(mutable.Map[String,Object]) = {

	var dict_aa = mutable.Map[String,Object] ()

	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t3051","和歌山",45178,"1998-7-24")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t3052","海南",42691,"1998-2-16")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t3053","橋本",32574,"1998-7-9")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t3054","有田",54756,"1998-5-2")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t3055","御坊",23875,"1998-9-9")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t3056","田辺",18296,"1998-6-30")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t3057","新宮",21578,"1998-4-22")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t3058","紀の川",17689,"1998-3-17")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t3059","岩出",43715,"1998-10-2")

	return	dict_aa
}

// ---------------------------------------------------------------------
def main( args: Array[String])
{
	println ("*** 開始 ***");

	def dict_aa = data_prepare_proc ()

	def factory = new SocketStoreClientFactory(
		new ClientConfig().setBootstrapUrls("tcp://localhost:6666"))
	def client:StoreClient[String,String] = factory.getStoreClient ("cities")

	voldemort_manipulate.dict_to_voldemort_proc (dict_aa,client)

	println ("*** 終了 ***")
}
// ---------------------------------------------------------------------
}
// ---------------------------------------------------------------------

File: voldemort/python/create/voldemort_create.py
#! /usr/bin/python
# -*- coding: utf-8 -*-
#
#	voldemort_read.py
#
#					Jan/27/2012
import voldemort
import json
import sys
#
sys.path.append ('/var/www/data_base/common/python_common')
from text_manipulate import dict_display_proc
from text_manipulate import dict_append_proc
from voldemort_manipulate import voldemort_to_dict_proc
# ------------------------------------------------------------
def	data_prepare_proc ():
	dict_aa = {} 
#
	dict_aa = dict_append_proc (dict_aa,'t3051',u'和歌山'.encode ('utf-8'),59624,'2003-4-30')
	dict_aa = dict_append_proc (dict_aa,'t3052',u'海南'.encode ('utf-8'),47237,'2003-5-10')
	dict_aa = dict_append_proc (dict_aa,'t3053',u'橋本'.encode ('utf-8'),21654,'2003-6-14')
	dict_aa = dict_append_proc (dict_aa,'t3054',u'有田'.encode ('utf-8'),83672,'2003-9-9')
	dict_aa = dict_append_proc (dict_aa,'t3055',u'御坊'.encode ('utf-8'),42391,'2003-8-4')
	dict_aa = dict_append_proc (dict_aa,'t3056',u'田辺'.encode ('utf-8'),35187,'2003-1-21')
	dict_aa = dict_append_proc (dict_aa,'t3057',u'新宮'.encode ('utf-8'),81246,'2003-7-23')
	dict_aa = dict_append_proc (dict_aa,'t3058',u'紀の川'.encode ('utf-8'),24784,'2003-10-26')
	dict_aa = dict_append_proc (dict_aa,'t3059',u'岩出'.encode ('utf-8'),75823,'2003-12-15')
#
	return	dict_aa
#
# ------------------------------------------------------------
print	("*** 開始 ***")
#
dict_aa = data_prepare_proc ()
#
client = voldemort.StoreClient('cities', [('localhost', 6666)])
#
for key in dict_aa.keys():
	unit_aa = dict_aa[key]
	json_str = json.dumps (unit_aa)
	client.put (key,json_str)
#
dict_aa = voldemort_to_dict_proc (client)
dict_display_proc (dict_aa)
#
print	("*** 終了 ***")
# ------------------------------------------------------------
